typed/merge.go

```diff
@@ -172,72 +172,83 @@ func (w *mergingWalker) visitListItems(t *schema.List, lhs, rhs value.List) (err
 	}
 	out := make([]interface{}, 0, int(math.Max(float64(rLen), float64(lLen))))
 
-	lhsOrder := make([]fieldpath.PathElement, 0, lLen)
-
-	// First, collect all LHS children.
-	observedLHS := fieldpath.MakePathElementValueMap(lLen)
-	if lhs != nil {
-		for i := 0; i < lhs.Length(); i++ {
-			child := lhs.At(i)
+	createPathElementsValues := func(name string, list value.List) ([]fieldpath.PathElement, fieldpath.PathElementValueMap, ValidationErrors) {
```

Comment:
Can we move the two anonymous function to actual functions? I'd rather avoid having loose function like these, especially in a big piece of code here where they might capture things by accident and make the semantics much harder to grasp.